### PR TITLE
Make elastic pool more generic :ok_hand:

### PR DESCRIPTION
**Pull request by michalwski**
This PR addresses review comment by @fenek https://github.com/esl/MongooseIM/pull/2079#pullrequestreview-157352574

### CODE REVIEW

**Comment by mongoose-im** [comment]
[5650.1](https://travis-ci.org/esl/MongooseIM/jobs/433433578) / Erlang 19.3 / small_tests / fd9ebdf27b3982cc8b4eb06b7824734330c7fbde
Reports [root](http://esl.github.io/mongooseim-ct-reports/s3_reports.html?prefix=PR/2087/5650/small_tests.19.3) / [small](https://mongooseim-ct-results.s3-eu-west-1.amazonaws.com/PR/2087/5650/small_tests.19.3/small/ct_run.mongooseim@localhost.2018-09-26_11.04.47/lib.mongooseim.logs/run.2018-09-26_11.04.48/suite.log.html)

**Comment by mongoose-im** [comment]
[5655.1](https://travis-ci.org/esl/MongooseIM/jobs/433693024) / Erlang 19.3 / small_tests / 78cecb482b9c11cacb8311229f510169c732adc6
Reports [root](http://esl.github.io/mongooseim-ct-reports/s3_reports.html?prefix=PR/2087/5655/small_tests.19.3) / [small](https://mongooseim-ct-results.s3-eu-west-1.amazonaws.com/PR/2087/5655/small_tests.19.3/small/ct_run.mongooseim@localhost.2018-09-26_19.09.58/lib.mongooseim.logs/run.2018-09-26_19.09.58/suite.log.html)

**Comment by mongoose-im** [comment]
[5658.1](https://travis-ci.org/esl/MongooseIM/jobs/433947706) / Erlang 19.3 / small_tests / 0a164acf3c97d0002af2898ba8b2db149b45a5f0
Reports [root](http://esl.github.io/mongooseim-ct-reports/s3_reports.html?prefix=PR/2087/5658/small_tests.19.3) / [small](https://mongooseim-ct-results.s3-eu-west-1.amazonaws.com/PR/2087/5658/small_tests.19.3/small/ct_run.mongooseim@localhost.2018-09-27_07.26.59/lib.mongooseim.logs/run.2018-09-27_07.27.00/suite.log.html)

File: doc/advanced-configuration/outgoing-connections.md [hunk 1]
```diff
@@ -213,20 +213,21 @@ A connection pool to ElasticSearch can be configured as follows:
 
 ```erlang
 {outgoing_pools, [
- {elastic, global, elasticsearch, [], [{host, "localhost"}]}
+ {elastic, global, default, [], [{host, "localhost"}]}
 ]}.
 ```
 
 MongooseIM uses [inaka/tirerl](https://github.com/inaka/tirerl) library to communicate with ElasticSearch.
 This library starts a pool of workers on its own so the following options are not configurable via `WPoolOpts`:
```
Comment (fenek) [inline doc/advanced-configuration/outgoing-connections.md]: Is it still valid, given the changes in `mongoose_wpool_elastic`? AFAIU now we're creating the pool on our own.

File: src/elasticsearch/mongoose_elasticsearch.erl [hunk 2]
```diff
@@ -40,7 +40,7 @@
 
 -include("mongoose.hrl").
 
--define(POOL_NAME, elasticsearch).
+-define(POOL_NAME, mongoose_wpool:make_pool_name(elastic, global, default)).
```
Comment (fenek) [inline src/elasticsearch/mongoose_elasticsearch.erl]: We need to make it more flexible later. For now it is good enough.

File: doc/advanced-configuration/outgoing-connections.md [hunk 3]
```diff
@@ -213,20 +213,21 @@ A connection pool to ElasticSearch can be configured as follows:
 
 ```erlang
 {outgoing_pools, [
- {elastic, global, elasticsearch, [], [{host, "localhost"}]}
+ {elastic, global, default, [], [{host, "localhost"}]}
```
Comment (fenek) [inline doc/advanced-configuration/outgoing-connections.md]: I would add an explicit note that currently only one `elastic` pool is supported.

File: doc/advanced-configuration/outgoing-connections.md (hunk 1 above)
Comment (fenek) [reply doc/advanced-configuration/outgoing-connections.md]: It's true, however "starts a pool of workers on its own" is not exactly the direct reason anymore.

Comment (michalwski) [reply doc/advanced-configuration/outgoing-connections.md]: True that :)

**Pull request merged**